### PR TITLE
Support endless operator methods

### PR DESCRIPTION
**Pull request by etiennebarrie**

### CODE REVIEW

File: spec/indent/method_definitions_spec.rb [hunk 1]
```diff
@@ -74,6 +74,51 @@ def foo(
...
+        def %(other) = to_i % other
+        def &(other) = to_i & other
+        def ^(other) = to_i ^ other
+        def >>(other) = to_i >> other
+        def <<(other) = to_i << other
+        def ==(other) = to_r == other
+        def !=(other) = to_r != other
+
+        def ===(other) = to_r === other # didn't need a change, but also broken
+        # def foo===(other) = fixme
```
Comment (AndrewRadev) [inline spec/indent/method_definitions_spec.rb:92]: The test seems to fail for me at this point and continue to fail down the line, does it pass for you? If so, there might be some local setting that needs to be ignored in the regex.

**Comment by AndrewRadev** [comment]
Thank you for your contribution, I didn't consider operator definitions when I worked on endless indenting. The test seems to fail for me at some point, does it pass for you or is the PR incomplete?

File: spec/indent/method_definitions_spec.rb (hunk 1 above)
Comment (etiennebarrie) [reply spec/indent/method_definitions_spec.rb:92]: Sorry I think I just hadn't finished the work 😬 

The comments meant that I didn't need to change anything for these to work, but apparently after rebasing I did need to handle these cases specifically in the regex.

fixme meant they didn't work, but I think I was just missing the `.` to define singleton methods instead.

**Comment by etiennebarrie** [comment]
The PR was incomplete, sorry. The test should now pass.

**Comment by AndrewRadev** [comment]
Thanks, I can confirm the tests are passing, and the regex looks sensible. I'll merge this and we'll see if any issues come up. 👍

**Pull request merged**